pkg/deploy/dev-workspace/dev_workspace_syncer.go

```diff
@@ -180,12 +180,41 @@ func syncDwDeployment(deployContext *deploy.DeployContext) (bool, error) {
 	}
 
 	devworkspaceControllerImage := util.GetValue(deployContext.CheCluster.Spec.DevWorkspace.ControllerImage, deploy.DefaultDevworkspaceControllerImage(deployContext.CheCluster))
-	devWorkspaceController := deployment.Spec.Template.Spec.Containers[0]
-	devWorkspaceController.Image = devworkspaceControllerImage
-	for _, env := range devWorkspaceController.Env {
+	deployment.Spec.Template.Spec.Containers[0].Image = devworkspaceControllerImage
+	deployment.Spec.Template.Spec.Containers[1].Image = deploy.DefaultImageName(deployContext.CheCluster, "gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0")
```

Comment:
There's no guarantee that the containers order will stay the same in the future, so this could break unexpectedly. Perhaps a safer bet is something like

```suggestion	var devworkspaceControllerContainer *corev1.Container
	var rbacProxyContainer *corev1.Container
	for idx, container := range deployment.Spec.Template.Spec.Containers {
		switch container.Name {
		case "kube-rbac-proxy":
			rbacProxyContainer = &deployment.Spec.Template.Spec.Containers[idx]
		case "devworkspace-controller":
			devworkspaceControllerContainer = &deployment.Spec.Template.Spec.Containers[idx]
		}
	}
	devworkspaceControllerContainer.Image = devworkspaceControllerImage
	rbacProxyContainer.Image = deploy.DefaultImageName(deployContext.CheCluster, "gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0")
	for idx, env := range devworkspaceControllerContainer.Env {
		switch env.Name {
		case "RELATED_IMAGE_devworkspace_webhook_server":
			devworkspaceControllerContainer.Env[idx].Value = devworkspaceControllerImage
		case "RELATED_IMAGE_kube_rbac_proxy":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_project_clone":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_plugin_redhat_developer_web_terminal_4_5_0":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_web_terminal_tooling":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_pvc_cleanup_job":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_async_storage_server":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		case "RELATED_IMAGE_async_storage_sidecar":
			devworkspaceControllerContainer.Env[idx].Value = deploy.DefaultImageName(deployContext.CheCluster, env.Value)
		}
	}
```
(note this includes the env var for loop below)